### PR TITLE
Add factoring func

### LINKED ISSUE #3: Factoring
Example Problem:
x^2+x-6
Example Solution:
(x-2)(x+3)

### PR DESCRIPTION
**Pull request by drobiu**
Added factoring function generator to the main python file

Closes #3

### CODE REVIEW

**Comment by drobiu** [comment]
Can you add a hacktoberfest-accepted label to this pull request and a hacktoberfest label to the repo? The hacktoberfest-accepted is only for the pull requests.

File: mathgenerator/mathgen.py [hunk 1]
```diff
@@ -216,6 +216,28 @@ def multiplyIntToMatrix22(maxMatrixVal = 10, maxRes = 100):
     solution = f"[[{a*constant},{b*constant}],[{c*constant},{d*constant}]]"
     return problem, solution
 
+def factoringFunc(range_x1 = 10, range_x2 = 10):
+  x1 = random.randint(-range_x1, range_x1)
+  x2 = random.randint(-range_x2, range_x2)
+  def intParser(int):
```
Comment (RobertWetzler) [inline mathgenerator/mathgen.py]: int overrides the python keyword int, consider renaming

**Comment by lukew3** [comment]
@drobiu When testing your implementation, the problem `x^2+x` was generated with a solution `(x)(x)`. Please correct this before I merge. Note that similar problems like `x^2+8x` were generated with the correct solution `(x)(x+8)`.

**Comment by lukew3** [comment]
Thanks for pointing out the hacktoberfest label issue on the repo.

**Review by lukew3 — DISMISSED** [review]
Fix (x)(x) solution issue mentioned in earlier comment.

**Pull request merged**